<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="penOptionsContainer" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>162</width>
      <height>34</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="labelCurrentPenWidth">
       <property name="text">
        <string>线宽：2</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="sliderPenWidth">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>20</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="fileToolBar">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSaveAs"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionClearCanvas"/>
   <addaction name="actionGroup"/>
   <addaction name="actionUngroup"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSelectTool"/>
   <addaction name="actionDrawFreehand"/>
   <addaction name="actionDrawLine"/>
   <addaction name="actionDrawRectangle"/>
   <addaction name="actionDrawEllipse"/>
   <addaction name="actionDrawStar"/>
   <addaction name="actionAiDraw"/>
  </widget>
  <widget class="QToolBar" name="toolBar_2">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionChangeColor"/>
   <addaction name="actionChangeFillColor"/>
  </widget>
  <widget class="QToolBar" name="toolBar_3">
   <property name="windowTitle">
    <string>toolBar_3</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNormalEraser"/>
   <addaction name="actionStrokeEraser"/>
   <addaction name="actionDraggingStrokeEraser"/>
  </widget>
  <action name="actionSelectTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/arrow_selector.svg</normaloff>:/icons/icons/arrow_selector.svg</iconset>
   </property>
   <property name="text">
    <string>选择</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;选择工具 (Esc)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;点击以选中画布上的图形，或拖动以移动选中的图形。 &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDrawLine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/line.svg</normaloff>:/icons/icons/line.svg</iconset>
   </property>
   <property name="text">
    <string>直线</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;绘制直线 (L)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;在画布上点击并拖动以画出一条直线段。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDrawRectangle">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/square.svg</normaloff>:/icons/icons/square.svg</iconset>
   </property>
   <property name="text">
    <string>矩形</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;绘制矩形 (R)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;在画布上点击并拖动以画出一个矩形。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionChangeColor">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/color.svg</normaloff>:/icons/icons/color.svg</iconset>
   </property>
   <property name="text">
    <string>颜色</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;更改边框颜色&lt;/span&gt;&lt;/p&gt;&lt;p&gt;点击选择后续绘制图形的边框颜色。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDrawFreehand">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/freehand_pen.svg</normaloff>:/icons/icons/freehand_pen.svg</iconset>
   </property>
   <property name="text">
    <string>自由画笔</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;自由画笔 (P)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;在画布上按住并拖动以自由绘制曲线。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionStrokeEraser">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/clicking_eraser.svg</normaloff>:/icons/icons/clicking_eraser.svg</iconset>
   </property>
   <property name="text">
    <string>点击笔画橡皮擦</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;笔画擦除（点击）&lt;/span&gt;&lt;/p&gt;&lt;p&gt;点击图形的任意部分以删除整个图形。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDraggingStrokeEraser">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/dragging_eraser.svg</normaloff>:/icons/icons/dragging_eraser.svg</iconset>
   </property>
   <property name="text">
    <string>拖动笔画橡皮擦</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;笔画擦除（拖动）&lt;/span&gt;&lt;/p&gt;&lt;p&gt;按住并拖动，鼠标划过的所有图形都将被删除。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionNormalEraser">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/normal_eraser.svg</normaloff>:/icons/icons/normal_eraser.svg</iconset>
   </property>
   <property name="text">
    <string>普通橡皮擦</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;橡皮擦&lt;/span&gt;&lt;/p&gt;&lt;p&gt;按住并拖动以擦除画布内容（使用背景色绘制），粗细可调。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionChangeFillColor">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/inner_color.svg</normaloff>:/icons/icons/inner_color.svg</iconset>
   </property>
   <property name="text">
    <string>填充颜色</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;更改填充颜色&lt;/span&gt;&lt;/p&gt;&lt;p&gt;点击选择后续绘制图形的填充颜色（对闭合图形有效）。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/undo.svg</normaloff>:/icons/icons/undo.svg</iconset>
   </property>
   <property name="text">
    <string>撤销</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;撤销 (Ctrl+Z)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;取消上一步操作。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/redo.svg</normaloff>:/icons/icons/redo.svg</iconset>
   </property>
   <property name="text">
    <string>重做</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;重做 (Ctrl+Shift+Z)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;恢复上一步已撤销的操作。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/save.svg</normaloff>:/icons/icons/save.svg</iconset>
   </property>
   <property name="text">
    <string>另存为...</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;另存为... (Ctrl + S)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;将当前画布内容保存为图像文件。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionClearCanvas">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/clear.svg</normaloff>:/icons/icons/clear.svg</iconset>
   </property>
   <property name="text">
    <string>清空画布</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;清空画布&lt;/span&gt;&lt;/p&gt;&lt;p&gt;清除画布上所有已绘制的图形，此操作可撤销。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/file_open.svg</normaloff>:/icons/icons/file_open.svg</iconset>
   </property>
   <property name="text">
    <string>打开图片...</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;打开图片 (Ctrl + O)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;加载一张图片作为画布背景。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDrawEllipse">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/ellipse.svg</normaloff>:/icons/icons/ellipse.svg</iconset>
   </property>
   <property name="text">
    <string>椭圆</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;绘制椭圆 (E)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;在画布上点击并拖动以画出一个椭圆。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionDrawStar">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/icons/icons/star.svg</normaloff>:/icons/icons/star.svg</iconset>
   </property>
   <property name="text">
    <string>五角星</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;绘制五角星(S)&lt;/span&gt;&lt;/p&gt;&lt;p&gt;在画布上点击并拖动以画出一个五角星。&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionAiDraw">
   <property name="text">
    <string>AI创作</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:700;&quot;&gt;AI创作&lt;/span&gt;&lt;/p&gt;&lt;p&gt;不知道画什么了？让AI帮你想想吧！&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionGroup">
   <property name="text">
    <string>编组</string>
   </property>
   <property name="toolTip">
    <string>将选中的多个图形编为一个组 (Ctrl+G)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionUngroup">
   <property name="text">
    <string>取消编组</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;将编组的图形解除为多个图形 (Ctrl+Shift+G)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+G</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
